lib/langchain/tool/weather.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Langchain::Tool
+  class Weather < Base
+    #
+    # A weather tool that gets current, forecast, or historical weather data
+    #
+    # Gem requirements:
+    #   gem "open-weather-ruby-client", "~> 0.3.0"
+    #
```

Comment:
Please write out Usage instructions. There's some examples of this in other files.

Reply:
done